examples/matching_engine/src/lib.rs

```diff
@@ -39,6 +39,18 @@ impl ServiceAbi for MatchingEngineAbi {
 /// Thus the quantity (also called _count_) is an `Amount`.
 ///
 /// When we have ask > bid then the winner for the residual cash is the liquidity provider.
+///
+/// We choose to force the price to be an integer u64. This is because the tokens are undivisible.
+/// In practice, this means that the value of token1 has to be much higher than the price of token0
+/// just as in a normal market where the price is in multiple of cents.
+///
+/// Implementing fractional price is more complicate. One complicacy is that we do not know to serialize
+/// fractions so that the serialization is in the same order as the fractions. But that is only a
+/// technical problem. If we want to keep the serialization order we can limit ourselves to fractions
+/// of the form say x / 100000.
+/// The next problem is that if we do the fractions, then the order can only be filled partially. And
+/// in a mathematical way, Euclidean divisions have to be done. This might be the right way to address
+/// the problem and could be of interest to other developers.
```

Comment:
Except if we also require that the token1 count is always divisible by 100000. Then the division is always exact.

Remember that an `Amount` contains a u128 that represents the number of attotokens (10<sup>-18</sup>); so even if it had to be divisible by 100000, or even a million, the divisibility for token1 amounts would still be 10<sup>-12</sup> (picotokens).

(Since tokens _are_ divisible, maybe `count` is also not the ideal name. Maybe just `amount`?)

Reply:
Well, if token1 is always divisible by 10000, why not divide it by 10000? Otherwise, this is a requirement on the input that feels odd and makes for not very nice code.
According to your own words token1 price has to be order of magnitude of token0. Then why not use a token0 that is an attotoken?


I can of course rename to amount.